ci/jenkins_build

```diff
@@ -50,9 +67,164 @@ docker run --rm \
+  cat /dev/urandom \
+    | tr -dc 'a-z0-9' \
+    | head -c 8 \
+    || true
+}
+
+function delete_namespace() {
+  if [[ "$namespace_deleted" == false ]]; then
+    if [[ -n $deployer_name ]]; then
+      announce "Collecting logs for deployer"
+      deployer_logs="$(kubectl logs "jobs/$deployer_name" \
+          --namespace="$NAMESPACE" \
+          --tail 9999 \
+          || echo "ERROR Failed to get logs for deployer $deployer_name")"
+      echo "$deployer_logs"
+      deployer_name=""
+    fi
+
+    if [[ "$display_manifests" == true ]]; then
+      announce "Displaying manifests from namespace \"$NAMESPACE\""
+      kubectl get all --namespace "$NAMESPACE" --output=yaml
+    fi
+
+    announce "Deleting namespace \"$NAMESPACE\""
+    kubectl delete namespace "$NAMESPACE" --ignore-not-found
+    namespace_deleted="true"
+  fi
+}
+
+function clean_and_exit() {
+  if [[ -n "$1" ]]; then
+    final_result "ERROR $1"
+  fi
+  delete_namespace
+  exit 1
+}
+
+function wait_for_deletion() {
+  kind="$1"
+  timeout="$2"
+  start_time=$(date +%s)
+  poll_interval=4
+
+  while true; do
+    # Everything under the namespace needs to be removed after app/uninstall
+    echo "INFO Checking if $kind were deleted"
+    resources=$(kubectl get $kind \
+      --namespace="$NAMESPACE" \
+      -o=json \
+      | jq -r '.items[] | "\(.kind)/\(.metadata.name)"')
+
+    res_count=$(echo $resources | wc -w)
+
+    if [[ "$res_count" -eq 0 ]]; then
+      break
+    else
+      # Ignore service account default
+      if [[ "$resources" = "ServiceAccount/default" ]]; then
+        break
+      else
+        echo "INFO Remaining: $res_count"
+        echo "INFO $resources"
+
+        elapsed_time=$(( $(date +%s) - $start_time ))
+        if [[ "$elapsed_time" -gt "$timeout" ]]; then
+          exit 1
+        fi
+
+        sleep "$poll_interval"
+      fi
+    fi
+  done
+}
+
+announce "Running build and tests..."
 pushd ../
   trap popd EXIT
-  ./build.sh
+
+  TEST_ID="$(random_string)"
+  export NAME="conjur"
+  export NAMESPACE="$NAME-$TEST_ID"
+  announce "Creating namespace $NAMESPACE"
+  kubectl create namespace "$NAMESPACE"
+  namespace_deleted=false
+  display_manifests=true
+
+  announce "Deploying Conjur application"
+  ./build.sh --persist
+
+  announce "Waiting for deployer to complete..."
+  deployer_name="${NAME}-deployer"
+  start_time=$(date +%s)
+  poll_interval=4
+  while true; do
+    deployer_status=$(kubectl get "jobs/$deployer_name" \
+                      --namespace="$NAMESPACE" -o=json | \
+                      jq '.status' || echo "{}")
+    failure=$(echo $deployer_status | jq '.failed // empty')
+    if [ -n "$failure" ]; then
+      clean_and_exit "Deployer failed"
+    fi
+
+    success=$(echo $deployer_status | jq '.succeeded // empty')
+    if [ -n "$success" ]; then
+      echo "INFO Deployer job succeeded"
+      break
+    fi
+
+    elapsed_time=$(( $(date +%s) - $start_time ))
+    printf "Elapsed ${elapsed_time}s\r"
```

Comment:
There's way too much mixing of printing methods: `echo`, `printf`, `announce`, and `final_result` you should try to cu this list at least in half.